### PR TITLE
use `round()` in `transform`s

### PR DESCRIPTION
**Pull request by mayank99**
## Changes

This PR helps with #1551 by rounding some percentage values to the nearest pixel, using the brand-new [`round()`](https://developer.mozilla.org/en-US/docs/Web/CSS/round) CSS function. This is mainly in `transform`s in `dialog` and `expandable-block`. And for browsers that don't support `round()`, the unrounded fallback will continue to be used.

Unrelated to the `round()` change but might possibly help towards #1551: simplified `information-panel` to remove `transform: translate(0)`.

I don't think this PR is the "ultimate" fix, so I'm not linking the issue to be closed. But it's a small net improvement nonetheless.

**Implementation notes**:
- I had to use `Round()` instead of `round()` to disambiguate from the Sass [`round()`](https://sass-lang.com/documentation/modules/math/#round) function.
- I had to add `@supports` blocks which would normally be unnecessary, because `lightningcss` was trying to "minify" the CSS by removing [repeated instances of the same property](https://github.com/iTwin/iTwinUI/pull/2064/commits/ad9fb59e103a1f5ac4d370ab1609f798123f8731).

## Testing

I tried to test in my Windows VM with some of the sandboxes linked in #1551. It was somewhat difficult to reproduce the issues in the first place, but I did see an improvement in at least one situation:

<details><summary>Screenshots</summary>

| Before | After |
| --- | --- |
| blurry text in expandable-block inside dialog | sharp text in same expandable-block inside same dialog |

</details> 

Further testing on a real Windows machine would be useful. (Make sure to update to Chrome 125)

## Docs

Added changesets.

### CODE REVIEW

**Review by Ben-Pusey-Bentley — APPROVED** [review]
Tested this PR out with a couple of the playgrounds from the issue, and there is a noticeable improvement.
Before:
image
After:
image
Before:
image
After:
image

File: .changeset/chilled-pugs-promise-2.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@itwin/itwinui-react': patch
+---
+
+Adjusted `Dialog` and `ExpandableBlock` to round some CSS `transform` values. This helps avoid blurry text on Windows in some cases.
```
Comment (rohan-kadkol) [inline .changeset/chilled-pugs-promise-2.md:5]: nit/optional: If this PR also led (or could lead) to improvements in the `InformationPanel` too, should it also be mentioned in another changeset (since it's unrelated to the rounded transform change)?

**Review by rohan-kadkol — APPROVED** [review]

File: .changeset/chilled-pugs-promise-2.md (hunk 1 above)
Comment (mayank99) [reply .changeset/chilled-pugs-promise-2.md:5]: I don't think it's worth mentioning, it's a "just in case" type of thing.

**Comment by Ben-Pusey-Bentley** [comment]
> @Ben-Pusey-Bentley Just for extra surety, could you share the code you used for testing? And could you try to keep all other things constant between the Before vs After? (e.g. same theme, same zoom level, same window size, etc)

Okay, for this code: 
```import './styles.css';
import React, { useCallback, useMemo, useState } from 'react';

import {
  CellProps,
  CellRendererProps,
  Column,
  IdType,
  Row,
  TableInstance,
  TableState,
} from 'react-table';

import {
  Checkbox,
  Code,
  InputGroup,
  DropdownMenu,
  MenuItem,
  IconButton,
  Table,
  Leading,
  tableFilters,
  TableFilterValue,
  TableProps,
  Tooltip,
  DefaultCell,
  EditableCell,
  TablePaginator,
  TablePaginatorRendererProps,
  ActionColumn,
  Anchor,
  SelectionColumn,
  ExpanderColumn,
  Input,
  Radio,
  ProgressRadial,
  useTheme,
  BaseFilter,
  Select,
  Tabs,
  Tab,
  Modal,
  Button,
  ModalContent,
} from '@itwin/itwinui-react';

type tableType = {
  id: string;
  name: string;
  description: string;
};

export default function App() {
  const [activeTab, setActiveTab] = React.useState(0);
  const [sort, setSort] = React.useState('');
  const [visualSort, setVisualSort] = React.useState('');

  const [isOpen, setIsOpen] = React.useState(false);
  const onClose = () => {
    setIsOpen(false);
  };

  const sortTypes = () => ['Custom', 'Recent', 'Name', 'Description'];

  const sortOptions = useMemo(
    () => (disabledValue?: string) =>
      sortTypes().map((type: string) => ({
        value: type,
        label: type,
        disabled: type === disabledValue,
      })),
    [],
  );

  const sortItems = useMemo(
    () => (data: tableType[], sortType: string) => {
      switch (sortType) {
        case sortTypes()[2]:
          return [...data].sort((a, b) => a.name.localeCompare(b.name));
        case sortTypes()[3]:
          return [...data].sort((a, b) =>
            a.description.localeCompare(b.description),
          );
        default:
          return data;
      }
    },
    [],
  );

  const filterEvenItems = (data: tableType[], filter: boolean) =>
    data.filter((d) => (filter ? Number(d.id) % 2 === 0 : true));

  const customSortLabel = sortTypes()[0];
  const globalSortOptions = sortOptions(customSortLabel);

  const sortWithCaseIgnore = React.useCallback(
    (
      rowA: Row<tableType>,
      rowB: Row<tableType>,
      columnId: IdType<tableType>,
    ) => {
      const nameA: string = rowA.values[columnId];
      const nameB: string = rowB.values[columnId];
      return nameA.localeCompare(nameB, undefined, {
        sensitivity: 'base', // case and accent marks ignored
      });
    },
    [],
  );

  const translatedLabels = React.useMemo(
    () => ({
      filter: 'Apply',
      clear: 'Clear',
      from: 'From',
      to: 'To',
    }),
    [],
  );

  const handleChange = (value: string) => {
    setVisualSort(value);
    setSort(value);
  };

  const onSort = useCallback(
    (state: TableState) => {
      const isSorted = state.sortBy.length !== 0;

      if (isSorted) setVisualSort(customSortLabel);
      else setVisualSort(sort);
    },
    [sort],
  );

  const columns = useMemo(
    (): Column<tableType>[] => [
      {
        id: 'id',
        Header: 'ID (Sorts Desc First)',
        accessor: 'id',
        sortDescFirst: true,
      },
      {
        id: 'name',
        Header: 'Name (Sorts Asc First)',
        accessor: 'name',
        fieldType: 'text',
        sortType: sortWithCaseIgnore,
        filter: 'text',
        Filter: tableFilters.TextFilter(translatedLabels),
      },
      {
        id: 'description',
        Header: 'Description',
        accessor: 'description',
        fieldType: 'text',
        sortType: sortWithCaseIgnore,
        filter: 'text',
        Filter: tableFilters.TextFilter(translatedLabels),
        maxWidth: 200,
        // disableSortBy: true
      },
    ],
    [],
  );
  const data = useMemo(
    () => [
      {
        id: '1',
        name: 'AA',
        description: 'Description4',
      },
      {
        id: '2',
        name: 'Name3',
        description: 'Description2',
      },
      {
        id: '3',
        name: 'aa',
        description: 'description1',
      },
      {
        id: '4',
        name: 'ba',
        description: 'description3',
      },
      {
        id: '5',
        name: 'fgdg',
        description: 'description56',
      },
      {
        id: '6',
        name: 'SFghfd',
        description: 'description78',
      },
      {
        id: '7',
        name: 'tyr',
        description: 'description33',
      },
      {
        id: '8',
        name: 'ds',
        description: 'description6',
      },
      {
        id: '9',
        name: 'asdw',
        description: 'description7',
      },
    ],
    [],
  );

  const duom = useMemo(
    () => filterEvenItems(sortItems(data, sort), activeTab === 1),
    [sort, data, activeTab, sortItems],
  );

  const onTabChange = (index: number) => {
    setActiveTab(index);
    if (visualSort === customSortLabel) {
      setVisualSort(sort);
    }
  };

  return (
    <div className='top-level'>
      <Button styleType='high-visibility' onClick={() => setIsOpen(true)}>
        Open Dialog
      </Button>
      <Modal
        className='modal-container'
        isOpen={isOpen}
        title='My modal'
        onClose={onClose}
        isDismissible
      >
        <ModalContent>
          <div className='appWindown'>
            <Tabs
              labels={[<Tab label='All' />, <Tab label='Some' />]}
              actions={[
                <Select
                  className='dropdown'
                  value={visualSort || globalSortOptions[1].value}
                  options={globalSortOptions}
                  onChange={(change) => handleChange(change || sortTypes()[1])}
                />,
              ]}
              activeIndex={activeTab}
              type='borderless'
              onTabSelected={onTabChange}
              wrapperClassName='listWrapper'
              tabsClassName='listTabs'
              contentClassName='listContent'
            >
              <Table
                className='table-list'
                columns={columns}
                data={duom}
                emptyTableContent='No data.'
                isSortable
                onSort={onSort}
              />
            </Tabs>
          </div>
        </ModalContent>
      </Modal>
    </div>
  );
}```
This is the before: 
![image](https://github.com/iTwin/iTwinUI/assets/141063311/6933717a-2489-48e0-a949-5095a0dd1cfa)
This is the after:
![image](https://github.com/iTwin/iTwinUI/assets/141063311/d466de12-ed6c-4b80-8527-9b9179eb410d)
For this code
```import React, { useState } from "react";
import {
  Text,
  Table,
  Code,
  Modal,
  ModalContent,
  ModalButtonBar,
  Button
} from "@itwin/itwinui-react";

export default function App() {
  const [modal1Open, setModal1Open] = useState(false);
  const [modal2Open, setModal2Open] = useState(false);

  const columns = [
    { id: "name", Header: "Name", accessor: "name" },
    { id: "description", Header: "Description", accessor: "description" },
    { id: "price", Header: "Price", accessor: "price" },
    { id: "quantity", Header: "Quantity", accessor: "quantity" }
  ];

  const data = [
    { name: "Carrot", description: "Orange", price: 0.79, quantity: 50 },
    { name: "Banana", description: "Yellow", price: 0.49, quantity: 30 },
    { name: "Apple", description: "Red", price: 1.29, quantity: 40 },
    { name: "Broccoli", description: "Green", price: 1.99, quantity: 20 },
    { name: "Strawberry", description: "Red", price: 3.49, quantity: 15 },
    { name: "Blueberry", description: "Blue", price: 4.99, quantity: 12 },
    { name: "Grape", description: "Purple", price: 2.79, quantity: 18 },
    { name: "Lemon", description: "Yellow", price: 0.69, quantity: 35 }
  ];

  return (
    <>
      <Text>
        <p>
          Dialog with <Code>translate</Code> has blurry text.
          <br />
          (Click the Open in New Window button in top right corner to see the
          effect more clearly.)
        </p>
      </Text>
      <Button styleType="high-visibility" onClick={() => setModal1Open(true)}>
        Open modal
      </Button>
      <Modal
        isOpen={modal1Open}
        title="Headers and row lines blurry"
        onClose={() => setModal1Open(false)}
      >
        <ModalContent>
          <Table columns={columns} data={data} emptyTableContent="No data." />
        </ModalContent>
        <ModalButtonBar>
          <Button onClick={() => setModal1Open(false)}>Close</Button>
        </ModalButtonBar>
      </Modal>

      <Text style={{ marginTop: "64px" }}>
        <p>
          Dialog without <Code>translate</Code> is clear.
        </p>
      </Text>
      <Button styleType="high-visibility" onClick={() => setModal2Open(true)}>
        Open modal
      </Button>
      <Modal
        isOpen={modal2Open}
        title="Headers and row lines clear"
        onClose={() => setModal2Open(false)}
        style={{ transform: "unset" }}
      >
        <ModalContent>
          <Table columns={columns} data={data} emptyTableContent="No data." />
        </ModalContent>
        <ModalButtonBar>
          <Button onClick={() => setModal2Open(false)}>Close</Button>
        </ModalButtonBar>
      </Modal>
    </>
  );
}
```
This is the before: 
image
This is the after:
image

**Comment by Ben-Pusey-Bentley** [comment]
> @Ben-Pusey-Bentley I don't see much of a difference (the "Before" text is not even blurry, at least in the screenshots). It would be nice to find a reproducible issue first. Try [this](https://github.com/iTwin/iTwinUI/issues/1551#issuecomment-1986311605) and [this](https://github.com/iTwin/iTwinUI/issues/1551#issuecomment-2010024165) and [this](https://github.com/iTwin/iTwinUI/issues/1551#issuecomment-2030049522)

This code shows the improvement better:
```
import * as React from 'react';
import {
  ExpandableBlock,
  Flex,
  Modal,
  ModalContent,
  Table,
  ToggleSwitch,
} from '@itwin/itwinui-react';

export default function App({
  isDvh,
  setIsDvh,
}: {
  isDvh: boolean;
  setIsDvh: (value: boolean) => void;
}) {
  const packageInfo = {
    '@bentley/documents-management': '2.1.7-dev.0',
    '@bentley/formsrenderer': '1.102.0',
    '@bentley/office365-collaboration-login-modal': '1.2.0',
    '@bentley/office365-collaboration-settings': '1.2.2',
    '@bentley/pw-api': '1.1.0-dev.1',
    '@bentley/pw-i18n': '1.0.4-dev.0',
    '@bentley/pwnxt-document-picker': '3.4.1',
    '@bentley/supply-chain-components': '0.0.3-beta.1',
    '@itwin/itwinui-react': '2.12.19',
  };

  return (
    <>
      <Modal isOpen={true} title="Connection info">
        <ModalContent>
          <Flex flexDirection="column" alignItems="stretch">
            <ToggleSwitch
              label="Use DVH units"
              defaultChecked={isDvh}
              onChange={() => setIsDvh(!isDvh)}
            />
            <ExpandableBlock
              title="Package versions"
              size="small"
              isExpanded={true}
            >
              <Table
                columns={[
                  { Header: 'Package', accessor: 'name' },
                  { Header: 'Version', accessor: 'version' },
                ]}
                emptyTableContent="No data"
                data={Object.entries(packageInfo).map(([name, version]) => ({
                  name,
                  version,
                }))}
                density="extra-condensed"
                style={{ maxHeight: '200px' }}
              />
            </ExpandableBlock>
          </Flex>
        </ModalContent>
      </Modal>
    </>
  );
}
```
Before: 
image

After:
image

The header text is still blurry in the after image, but the text inside of the table rows isn't blurry anymore.
